worker/docker_client.py

```diff
@@ -267,7 +272,10 @@ def start_container(self, bundle_path, uuid, command, docker_image,
             # This can cause problems if users expect to run as a specific user
             'User': '%s:%s' % (uid, gid),
         }
-        if self._use_nvidia_docker:
+        # Using nvidia-docker when GPU is not needed can have unintended
+        # side effects, so we only do so when GPUs are requested.
+        # TODO: Allocate the specified number of GPUs and isolate
+        if request_gpus > 0 and self._use_nvidia_docker:
```

Comment:
I don't know any specific examples, but according to [the nvidia-docker documentation](https://github.com/NVIDIA/nvidia-docker/wiki/Image-inspection):

> Mounting user-level driver libraries and device files clobbers the environment of the container, it should be done only when the container is running a GPU application. 

I have noticed that most users on the NLP instance have included the `--request-gpus` flag on their GPU jobs, so it might not be so bad? But I'm not opposed to removing this check until someone complains about their container being clobbered. It might never happen since nvidia-docker only modifies `/usr/local/nvidia` and the device files as far as I know.